src/kv.rs

```diff
@@ -320,171 +350,179 @@ impl fmt::Debug for KvPair {
+/// A struct for expressing ranges. This type is semi-opaque and is not really meant for users to
+/// deal with directly. Most functions which operate on ranges will accept any types which
+/// implement `Into<BoundRange>`.
 ///
 /// In TiKV, keys are an ordered sequence of bytes. This means we can have ranges over those
 /// bytes. Eg `001` is before `010`.
 ///
-/// This trait has implementations for common range types like `a..b`, `a..=b` where `a` and `b`
-/// `impl Into<Key>`. You could implement this for your own types.
+/// `Into<BoundRange>` has implementations for common range types like `a..b`, `a..=b` where `a` and `b`
+/// `impl Into<Key>`. You can implement `Into<BoundRange>` for your own types by using `try_from`.
+///
+/// Invariant: a range may not be unbounded below.
 ///
 /// ```rust
-/// use tikv_client::{KeyRange, Key};
+/// use tikv_client::{BoundRange, Key};
 /// use std::ops::{Range, RangeInclusive, RangeTo, RangeToInclusive, RangeFrom, RangeFull, Bound};
+/// # use std::convert::TryInto;
 ///
 /// let explict_range: Range<Key> = Range { start: Key::from("Rust"), end: Key::from("TiKV") };
-/// let from_explict_range = explict_range.into_bounds();
+/// let from_explict_range: BoundRange = explict_range.into();
 ///
 /// let range: Range<&str> = "Rust".."TiKV";
-/// let from_range = range.into_bounds();
+/// let from_range: BoundRange = range.into();
 /// assert_eq!(from_explict_range, from_range);
 ///
 /// let range: RangeInclusive<&str> = "Rust"..="TiKV";
-/// let from_range = range.into_bounds();
+/// let from_range: BoundRange = range.into();
 /// assert_eq!(
+///     from_range,
 ///     (Bound::Included(Key::from("Rust")), Bound::Included(Key::from("TiKV"))),
-///     from_range
 /// );
 ///
 /// let range_from: RangeFrom<&str> = "Rust"..;
-/// let from_range_from = range_from.into_bounds();
+/// let from_range_from: BoundRange = range_from.into();
 /// assert_eq!(
-///     (Bound::Included(Key::from("Rust")), Bound::Unbounded),
 ///     from_range_from,
-/// );
-///
-/// let range_to: RangeTo<&str> = .."TiKV";
-/// let from_range_to = range_to.into_bounds();
-/// assert_eq!(
-///     (Bound::Unbounded, Bound::Excluded(Key::from("TiKV"))),
-///     from_range_to,
-/// );
-///
-/// let range_to_inclusive: RangeToInclusive<&str> = ..="TiKV";
-/// let from_range_to_inclusive = range_to_inclusive.into_bounds();
-/// assert_eq!(
-///     (Bound::Unbounded, Bound::Included(Key::from("TiKV"))),
-///     from_range_to_inclusive,
-/// );
-///
-/// let range_full: RangeFull = ..;
-/// let from_range_full = range_full.into_bounds();
-/// assert_eq!(
-///     (Bound::Unbounded, Bound::Unbounded),
-///     from_range_full
+///     (Bound::Included(Key::from("Rust")), Bound::Unbounded),
 /// );
 /// ```
 ///
-/// **But, you should not need to worry about all this:** Many functions accept a `impl KeyRange`
+/// **But, you should not need to worry about all this:** Many functions accept a `impl Into<BoundRange>`
```

Comment:
```suggestion
/// **But, you should not need to worry about all this:** Most functions which operate
/// on ranges will accept any types  which implement `Into<BoundRange>`.
```

Reply:
This gets rid of "a impl" by repeating the more clearly written line from the first paragraph.